python/pyspark/ml/param/__init__.py

```diff
@@ -492,7 +493,7 @@ def _dummy() -> "Params":
         dummy.uid = "undefined"
         return dummy
 
-    def _set(self, **kwargs: Any) -> "Params":
+    def _set(self: P, **kwargs: Any) -> P:
```

Comment:
This is allow us to hint specific subclass of `Params`, that depends on the input.